data/bookkeeping/genesis.go

```diff
@@ -101,6 +101,51 @@ func (genesis Genesis) ID() string {
 	return string(genesis.Network) + "-" + genesis.SchemaID
 }
 
+// Hash is the genesis hash.
+func (genesis Genesis) Hash() crypto.Digest {
+	return crypto.HashObj(genesis)
+}
+
+// Balances returns the genesis account balances.
+func (genesis Genesis) Balances() (GenesisBalances, error) {
+	genalloc := make(map[basics.Address]basics.AccountData)
+	for _, entry := range genesis.Allocation {
+		addr, err := basics.UnmarshalChecksumAddress(entry.Address)
+		if err != nil {
+			return GenesisBalances{}, fmt.Errorf("cannot parse genesis addr %s: %w", entry.Address, err)
+		}
+
+		_, present := genalloc[addr]
+		if present {
+			return GenesisBalances{}, fmt.Errorf("repeated allocation to %s", entry.Address)
+		}
+
+		genalloc[addr] = entry.State
+	}
+
+	feeSink, err := basics.UnmarshalChecksumAddress(genesis.FeeSink)
+	if err != nil {
+		return GenesisBalances{}, fmt.Errorf("cannot parse fee sink addr %s: %w", genesis.FeeSink, err)
+	}
+
+	rewardsPool, err := basics.UnmarshalChecksumAddress(genesis.RewardsPool)
+	if err != nil {
+		return GenesisBalances{}, fmt.Errorf("cannot parse rewards pool addr %s: %w", genesis.RewardsPool, err)
+	}
+
+	return MakeTimestampedGenesisBalances(genalloc, feeSink, rewardsPool, genesis.Timestamp), nil
+}
+
+// Block computes the genesis block.
+func (genesis Genesis) Block() (Block, error) {
```

Comment:
not used

Reply:
We'd like to call this one from Indexer.

Reply:
call once on initialization? If calling often then the `Balances()` need to memoize its result